### PR TITLE
Document `FileData` class in docs 

### PR DESCRIPTION
**Pull request by hannahblair**
## Description

Documents the `FileData` class. We don't have an example for it in place - can someone provide a use case for this in a gradio demo plz? 

Closes: #8717

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

**Review by freddyaboulton — COMMENTED** [review]
The new `type='messages`` parameter in the chatbot allows you to return a `FileData` directly!

File: gradio/data_classes.py [hunk 1]
```diff
@@ -219,7 +220,21 @@ class FileDataDict(TypedDict):
     meta: dict
 
 
+@document()
 class FileData(GradioModel):
+    """
+    The FileData class is a subclass of Pydantic's BaseModel class that represents a file object within a Gradio interface. It is used to store file data and metadata when a file is uploaded.
```
[the remark this answers is not in the document]
Comment (freddyaboulton) [reply gradio/data_classes.py:226]: We can just say GradioModel I think - pydantic is just an implementation detail

**Review by aliabd — APPROVED** [review]
documentation looks good!

**Comment by abidlabs** [comment]
Thanks @hannahblair! The use case I had in mind was when you use [the Python Client with `download_files=False`](https://www.gradio.app/docs/python-client/client), then it returns a `FileData` object

**Review by abidlabs — APPROVED** [review]
LGTM thank so much for taking care of this @hannahblair!

**Comment by abidlabs** [comment]
Hmm interesting that the api info test is failing, let me take a look

**Comment by abidlabs** [comment]
Tweaked the `api_info()` method so that we still see a very readable `Client.view_api()`:

image

Should be all good now!

**Comment by abidlabs** [comment]
I'll go ahead and get this in so that we can cut down on the open PRs. Thanks @hannahblair!

**Pull request merged**